packages/lesswrong/server/repos/UltraFeedEventsRepo.ts

```diff
@@ -58,6 +58,49 @@ class UltraFeedEventsRepo extends AbstractRepo<'UltraFeedEvents'> {
 
     return unviewedItems.map((row: any) => row.documentId);
   }
+
+  /**
+   * Checks whether posts have been viewed by a user.
+   * Returns a Set of post IDs that have been viewed (either in UltraFeedEvents or ReadStatuses).
+   */
+  async getViewedPostIds(
```

Comment:
I don't like splitting this out into a separate query but I guess if some of the posts you're trying to run it on are from recombee you might not be able to get that info from them directly?

Reply:
Also, have you performance tested this query?  I'm not sure `AND "postId" = ANY($(postIds)::text[])` uses the relevant index correctly

Reply:
Yeah, this is for Recombee posts. I ran it on a bunch of heavy users and all results were < 100 ms